src/test/java/com/training/services/tests/VoyagesTest.java

```diff
@@ -74,12 +74,35 @@ public void testVoyageSailDurationValidate() {
     public void testVoyageShipCodeValidate() {
         Response voyageResponse = getResponse();
         logger.info("status->" + voyageResponse.getStatus());
-        logger.info("totalVoyagesCount->"+voyageResponse.getPayload().getVoyages().size());
+        logger.info("totalVoyagesCount->" + voyageResponse.getPayload().getVoyages().size());
         Assertions.assertThat(voyageResponse.getStatus()).isEqualTo("200")
                 .as(" status is not 200");
         List<Voyages> voyages = voyageResponse.getPayload().getVoyages();
         for (int i = 0; i < voyages.size(); i++) {
-         softAssert.assertThat(voyages.get(i).getShipCode()).isEqualTo("AL");
+            softAssert.assertThat(voyages.get(i).getShipCode()).isEqualTo("AL");
+        }
+    }
+
+    /**
+     * testVoyageMasterSailDateValidate:comparing sailDate with both master1sailDate and master2SailDate.
+     */
+    @Test
+    public void testVoyageMasterSailDateValidate() {
+        Response voyageResponse = getResponse();
+        logger.info("status--->" + voyageResponse.getStatus());
+        List<Voyages> voyages = voyageResponse.getPayload().getVoyages();
+        logger.info("totalVoyagesCount-->" + voyages.size());
+        Assertions.assertThat(voyageResponse.getStatus()).isEqualTo("200")
+                .as(" status is not 200");
+        for (int index = 0; index < voyages.size(); index++) {
+            String sailDate = voyages.get(index).getSailDate();
+            logger.info("sailDate->" + sailDate + " " + "index valuesS1-->" + index);
+            softAssert.assertThat(sailDate.equals(voyages.get(index).getMasterSailDate()
+                    .getMaster1SailDate())).isTrue()
+                    .as("sailDate and master1sailDate are not equal");
+            softAssert.assertThat(sailDate.equals(voyages.get(index).getMasterSailDate()
+                    .getMaster2SailDate())).isTrue()
+                    .as("sailDate and master2sailDate are not equal");
```

Comment:
lets also add the index to the assertion message telling the user the voyage no. i